archive/android_bootldr.ksy

```diff
@@ -0,0 +1,36 @@
+meta:
+  id: android_bootldr
+  title: Qualcomm Snapdragon (MSM) bootloader.img format
+  file-extension: img
+  tags:
+    - archive
+    - android
+  license: CC0-1.0
+  endian: le
+doc-ref: https://android.googlesource.com/device/lge/hammerhead/+/7618a7/releasetools.py
+seq:
+  - id: magic
+    contents: BOOTLDR!
+  - id: images
+    -orig-id: num_images
+    type: u4
+  - id: ofs_start
+    -orig-id: start_offset
+    type: u4
+  - id: bootloader_size
+    -orig-id: bootldr_size
+    type: u4
+  - id: img_info
+    type: img_info
+    repeat: expr
+    repeat-expr: images
```

Comment:
I know that the noun `info`(rmation) doesn't have a plural form in English, but the name of each repeated field should ideally be distinguishable from the singular form.

https://doc.kaitai.io/ksy_style_guide.html#attr-id:

> * For an array of objects (i.e. with repeat: something), use plural form — e.g. `files`, `transactions`.

The original identifier of this attribute doesn't matter, you can always preserve it in `-orig-id`. I wouldn't even mind if you use `img_infos`, regardless of what Standard English says about it. At least, there would be a clear distinction.

Or you can replace `info` with some noun that can be pluralized, e.g. `header`.

Besides that, making the subtype and attribute names different will make the KSC-generated code from this format spec usable in C# - when there is an attribute `img_info` with the same name as the subtype `img_info`, the C# codegen would be affected by https://github.com/kaitai-io/kaitai_struct/issues/86. In complex formats, this is somewhat tiresome to eliminate (so we usually don't even mention it as there are dozens of more serious issues), but dealing with this occurrence in this simple format makes sense I think.